### PR TITLE
[User management] Disabling user management in auth [DPP-827]

### PR DESCRIPTION
**Pull request by pbatko-da**

### CODE REVIEW

File: ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/interceptor/AuthorizationInterceptor.scala [hunk 1]
```diff
@@ -76,33 +79,43 @@ final class AuthorizationInterceptor(
   private[this] def resolveAuthenticatedUserRights(claimSet: ClaimSet): Future[ClaimSet] =
     claimSet match {
       case ClaimSet.AuthenticatedUser(userIdStr, participantId, expiration) =>
-        Ref.UserId.fromString(userIdStr) match {
-          case Left(err) =>
+        userManagementStoreO match {
+          case None =>
             Future.failed(
-              errorFactories.invalidArgument(None)(s"token $err")(errorLogger)
+              LedgerApiErrors.AuthorizationChecks.Unauthenticated
```
Comment (pbatko-da) [inline ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/interceptor/AuthorizationInterceptor.scala:85]: fyi @meiersi-da

**Review by meiersi-da — APPROVED** [review]
LGTM.

@mziolekda do we already set this feature flag to disabled when assembling the `pkv-participant` in the oem repo?

File: ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/interceptor/AuthorizationInterceptor.scala [hunk 2]
```diff
@@ -76,33 +79,43 @@ final class AuthorizationInterceptor(
...
+                          ledgerId = None,
+                          participantId = participantId,
+                          applicationId = Some(userId),
+                          expiration = expiration,
+                          resolvedFromUser = true,
+                        )
+                      )
+                  }
+            }
         }
```
Comment (kamil-da) [inline ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/interceptor/AuthorizationInterceptor.scala:119]: This amount of nesting made it a bit difficult to understand what's going on here. Would it be possible to split this into a few smaller methods?

**Comment by mziolekda** [comment]
@meiersi-da I am not exactly sure what you are asking about. But the user management is disabled by default in kv/app derivates.

File: ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/interceptor/AuthorizationInterceptor.scala (hunk 2 above)
Comment (pbatko-da) [reply ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/interceptor/AuthorizationInterceptor.scala:119]: Ok

**Review by mziolekda — APPROVED** [review]

**Pull request merged**